components/compliance-service/integration_test/reports_projects_update_test.go

```diff
@@ -1397,30 +1397,32 @@ func TestStartProjectUpdateWhenIDIsSent(t *testing.T) {
 	_, err := localSuite.ComplianceIngestServer.HandleEvent(context.Background(), event)
 	assert.NoError(t, err)
 
+	complete := false
 	// Wait for job to complete
-	for {
+	for !complete {
 		time.Sleep(time.Millisecond * 100)
-		if lastEventSent.Type.Name != automate_event_type.ProjectRulesUpdateStatus {
-			continue
-		}
 
-		if lastEventSent.Data.Fields["Completed"].GetBoolValue() {
-			break
+		for _, event := range eventsSent {
+			assert.Equal(t, event.Type.Name, automate_event_type.ProjectRulesUpdateStatus)
+			if event.Data.Fields["Completed"].GetBoolValue() {
+				complete = true
```

Comment:
Instead of using a boolean, could this be `break`?

Reply:
I tried that before. A `break` here will exit from the `for _, event := range eventsSent` for on line 1405. Here we want to break out of the `for` outside of that. 

But after looking at it again, I could just do a return 😄 .